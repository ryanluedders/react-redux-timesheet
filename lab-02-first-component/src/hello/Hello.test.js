import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Hello from './Hello';

describe('Hello World:', () => {

    it('renders without exploding', () => {
        expect(shallow(<Hello />)).toHaveLength(1);
    });

    it('should render with default test', () => {
        const component = shallow(<Hello />);

        expect(component).toIncludeText('Howdy');
        expect(component).toIncludeText('Partner');
    });

    it('should render with our props', () => {
        const component = shallow(<Hello friend="Fred" />);

        expect(component).toIncludeText('Howdy');
        expect(component).toIncludeText('Fred');
        expect(component).not.toIncludeText('Partner');
    })

    it('renders to match a snapshot', () => {
        const component = renderer.create(<Hello friend="Obi Wan" />);

        expect(component.toJSON()).toMatchSnapshot();
    })

});