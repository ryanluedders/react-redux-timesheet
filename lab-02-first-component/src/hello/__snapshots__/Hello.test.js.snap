// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders to match a snapshot 1`] = `
<div
  className="hello"
>
  <h1>
    Howdy!!
  </h1>
  <h2>
    Obi Wan
  </h2>
  <p>
    Congratulations!  You have created your first React component!
  </p>
</div>
`;
